File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/account/balance/db/DBBalance.java
package com.fintechplatform.ui.account.balance.db;

import com.raizlabs.android.dbflow.sql.language.SQLite;

/**
 * Created by ingrid on 13/02/18.
 */

public class DBBalance {
    public void deleteBalance() {
        SQLite.delete().from(Balance.class).execute();
    }

    public void saveBalance(Balance balance) {
        balance.save();
    }

    public Balance findBalance(String accountId) {
        return SQLite.select()
                .from(Balance.class)
                .where(Balance_Table.accountId.eq(accountId))
                .querySingle();
    }
}

File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/account/balance/db/DBBalance.kt
package com.fintechplatform.ui.account.balance.db

import com.raizlabs.android.dbflow.sql.language.SQLite

class DBBalance {
    fun deleteBalance() {
        SQLite.delete().from(Balance::class.java).execute()
    }

    fun saveBalance(balance: Balance) {
        balance.save()
    }

    fun findBalance(accountId: String): Balance? {
        return SQLite.select()
                .from(Balance::class.java)
                .where(Balance_Table.accountId.eq(accountId))
                .querySingle()
    }
}